rules/hmap.bzl

```diff
@@ -42,9 +42,15 @@ def _make_headermap_impl(ctx):
 
     # Write a file for *this* headermap, this is a temporary file
     input_f = ctx.actions.declare_file(ctx.label.name + "_input.txt")
-    all_hdrs = []
-    for provider in ctx.attr.hdrs:
-        all_hdrs += provider.files.to_list()
+    all_hdrs = list(ctx.files.hdrs)
+    for provider in ctx.attr.direct_hdr_providers:
+        if apple_common.Objc in provider:
+            all_hdrs += provider[apple_common.Objc].direct_headers
+        elif CcInfo in hdr_provider:
+            all_hdrs += provider[CcInfo].compilation_context.direct_headers
+        else:
+            fail("direct_hdr_provider %s must contain either 'CcInfo' or 'objc' provider" % provider)
+
     out = _make_headermap_input_file(ctx.attr.namespace, all_hdrs, ctx.attr.flatten_headers)
     ctx.actions.write(
```

Comment:
The `mappings` variable defined a few lines below is no longer being used, I think. It seems like it remains empty now.

Reply:
good catch! I'll delete the code referencing it